### ISSUE TITLE
Unhandled Exception: type 'Account' is not a subtype of type 'Order' of 'obj'

### ISSUE THREAD

**Issue by TomasWeg** [hivedb/hive#568]
I got this error when I try to save an `Account` to a box:
`
E/flutter (19165): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: type 'Account' is not a subtype of type 'Order' of 'obj'
E/flutter (19165): #0      OrderAdapter.write (package:distribuidora_griwert_distributors/application/data_source/adapters/order_adapter.dart)
package:distribuidora_griwert_distributors/…/adapters/order_adapter.dart:1
E/flutter (19165): #1      BinaryWriterImpl.write
package:hive/…/binary/binary_writer_impl.dart:316
E/flutter (19165): #2      BinaryWriterImpl.writeFrame
package:hive/…/binary/binary_writer_impl.dart:254
E/flutter (19165): #3      StorageBackendVm.writeFrames.<anonymous closure>
package:hive/…/vm/storage_backend_vm.dart:123
E/flutter (19165): #4      ReadWriteSync.syncWrite.<anonymous closure>
package:hive/…/vm/read_write_sync.dart:26
`
It seems like Hive is getting the wrong adapter. `Account`and `Order` are classes that have their corresponding adapter.
They are registered in the following order: `OrderAdapter(), AccountAdapter()`. If I invert the order, the exception is the same when I try to save an Order type.

Adapters are registered before `Hive.init()`

**Comment by TheMisir** [comment]
You've probably used same typeId for multiple types.

**Comment by kirill-plotnikov** [comment]
> You've probably used same typeId for multiple types.

Same problem. 
![image](https://user-images.githubusercontent.com/51074246/108618810-e430d080-7442-11eb-845a-2c846801aa12.png)

**Comment by kirill-plotnikov** [comment]
> > You've probably used same typeId for multiple types.
> 
> Same problem.
> ![image](https://user-images.githubusercontent.com/51074246/108618810-e430d080-7442-11eb-845a-2c846801aa12.png)

I hope this information helps someone. I reinstall the app - everything worked. I guess it was corrupted data. 

btw can i erase all the hive data from device with internal hive library methods?

**Comment by TheMisir** [comment]
> btw can i erase all the hive data from device with internal hive library methods?

Yes, there's a method called [`deleteFromDisk()`](https://pub.dev/documentation/hive/latest/hive/BoxBase/deleteFromDisk.html).

**Comment by mohammadne** [comment]
same problem

**Comment by kirill-plotnikov** [comment]
> same problem

Try to define adapter type when register it.
`Hive.registerAdapter<TypeOfAdapterHere>(adapter))`

**Comment by mohammadne** [comment]
> > same problem
> 
> Try to define adapter type when register it.
> `Hive.registerAdapter<TypeOfAdapterHere>(adapter))`

yes, that's it.
but is there any way to register list of `TypeAdapter` in place, like :
``` Dart
adapters.forEach((adapter) => Hive.registerAdapter(adapter));
```

**Comment by kirill-plotnikov** [comment]
> > > same problem
> > 
> > 
> > Try to define adapter type when register it.
> > `Hive.registerAdapter<TypeOfAdapterHere>(adapter))`
> 
> yes, that's it.
> but is there any way to register list of `TypeAdapter` in place, like :
> 
> ```dart
> adapters.forEach((adapter) => Hive.registerAdapter(adapter));
> ```

Maybe like that?
```
Map<dynamic, TypeAdapter> adapters = {
FirstType: FirstTypeAdapter,
SecondType: SecondTypeAdapter
}
adapters.forEach((k,v) => Hive.registerAdapter<k>(v));
```
but i doubt it will work)

**Comment by mohammadne** [comment]
> ``` Dart
> Map<dynamic, TypeAdapter> adapters = {
> FirstType: FirstTypeAdapter,
> SecondType: SecondTypeAdapter
> }
> adapters.forEach((k,v) => Hive.registerAdapter<k>(v));
> ```

will get :
```
The name 'k' isn't a type so it can't be used as a type argument.
```

**Comment by kirill-plotnikov** [comment]
> > ```dart
> > Map<dynamic, TypeAdapter> adapters = {
> > FirstType: FirstTypeAdapter,
> > SecondType: SecondTypeAdapter
> > }
> > adapters.forEach((k,v) => Hive.registerAdapter<k>(v));
> > ```
> 
> will get :
> 
> ```
> The name 'k' isn't a type so it can't be used as a type argument.
> ```

Yes, it was a stupid suggestion) Sorry.
